app/src/test/java/com/appunite/loudius/domain/PullRequestRepositoryImpTest.kt

```diff
@@ -85,28 +102,98 @@ class PullRequestRepositoryImpTest {
                         ),
                     ),
                 )
-
                 assertEquals(expected, actual)
             }
-        // TODO: Write tests with failure cases
+
+        @Test
+        fun `GIVEN incorrect pull request number WHEN get requested reviewers THEN return Unknown Error with 404 code`() =
+            runTest {
+                val pullRequestNumber = "incorrectPullRequestNumber"
+
+                val actual = repository.getRequestedReviewers(
+                    "example",
+                    "example",
+                    pullRequestNumber,
+                )
+
+                assertEquals(
+                    Result.failure<List<RequestedReviewersResponse>>(
+                        WebException.UnknownError(404, null)
+                    ),
+                    actual
+                )
+            }
     }
 
     @Nested
     inner class NotifyTest {
 
         @Test
-        fun `GIVEN correct values WHEN notify THEN return success result`() = runTest {
+        fun `GIVEN correct pull request number WHEN notifying THEN return success result`() = runTest {
+            val pullRequestNumber = "correctPullRequestNumber"
+
             val actual = repository.notify(
                 "exampleOwner",
                 "exampleRepo",
-                "correctPullRequestNumber",
+                pullRequestNumber,
                 "@ExampleUser",
             )
 
-            val expected = Result.success(Unit)
+            assertEquals(Result.success(Unit), actual)
+        }
 
-            assertEquals(expected, actual)
+        @Test
+        fun `GIVEN incorrect pull request number WHEN notifying THEN return success result`() = runTest {
+            coEvery {
+                repository.notify(any(), any(), any(), any())
+            } returns Result.failure(WebException.UnknownError(404, null))
+            val pullRequestNumber = "incorrectPullRequestNumber"
+
+            val actual = repository.notify(
+                "exampleOwner",
+                "exampleRepo",
+                pullRequestNumber,
+                "@ExampleUser",
+            )
+
+            assertEquals(
+                Result.failure<Unit>(WebException.UnknownError(404, null)),
+                actual
+            )
         }
-        // TODO: Write tests with failure cases
     }
+
+    @Nested
+    inner class GetPullRequestsForUserTest {
+        @Test
+        fun `GIVEN logged in user WHEN getting user's pull requests THEN return pull requests`() =
+            runTest {
+                coEvery { userDataSource.getUser() } returns Result.success(
```

Comment:
suggestion: I think by default the mock recurs correct data. You don't need to mock it.

Reply:
I have added a big commit now, where I introduce to tests 'currentUser'. It might get hard to control it/make them work together when we have many different fakes and sometimes we use different names for the current user, e.g. `user1`, `exampleUser`.